src/app/view/components/_common/dot-icon-button/dot-icon-button.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, Input, EventEmitter, Output, HostListener } from '@angular/core';
+
+/**
+ * The DotIconButtonComponent is a round button which
+ * reuses the dot-icon component
+ * @export
+ * @class DotIconButtonComponent
+ */
+@Component({
+    selector: 'dot-icon-button',
+    styleUrls: ['./dot-icon-button.component.scss'],
+
+    template: `<div class="dot-icon-button
```

Comment:
This seems to complex to not being in their own template file.

The click event should be in a `<button>` not a `div`, semantics matter.